### DISCUSSION TITLE
Whether thermal expansion and creep can be used together？

### DISCUSSION THREAD

**Discussion by Moyiqi** [idaholab/moose#24131] [Q&A General] [answered]
Hello everyone, may I ask a question?
```
    # [stress]
    #   type = ComputeFiniteStrainElasticStress
    # []
    [./thermal_strain1]
      type = ComputeThermalExpansionEigenstrain
      thermal_expansion_coeff = 1.243e-11
      stress_free_temperature = 230
      temperature = temperature
      eigenstrain_name = eigenstrain
    [../]
    [./density1]
    type = Density
    density = 2736
    [../]
    [./radial_return_stress]
      type = ComputeMultipleInelasticStress
      inelastic_models = 'powerlawcrp plas'
    [../]
    [./powerlawcrp]
      type = PowerLawCreepStressUpdate
      coefficient = 3
      n_exponent = 3
      m_exponent = 2
      activation_energy = 0.2
    [../]
    [./plas]
      type = IsotropicPlasticityStressUpdate
      hardening_constant = 0.5
      yield_stress =300
    [../]
```
When I use block [stress] , another block [radial_return_stress]  can't be used. So I had to go into hiding.
Can I use together?

Lily

**Comment by GiudGiud** [comment, marked as answer]
Hello

No you cannot define multiple stress models on the same blocks (subdomains), unless you use either the "suffix" parameter to disambiguate them. 

Fyi:
> So I had to go into hiding.

You had to comment it out. Going into hiding means something else.

**Reply by Moyiqi** [reply]
We want to add creep, plastic deformation, elastic deformation and thermal expansion into the equation. Is there any similar code available？
Lily

**Reply by GiudGiud** [reply]
I think the ComputeMultipleInelasticStress is the way to go, which you are already using. 

Others may pitch in for alternatives

**Reply by Moyiqi** [reply]
Thank you for your reply, may I ask you another question?
When I add '''prop_getter_suffix=prop_name''' in '''ComputeMultipleInelasticStress'''
  I get an error '''Material property 'mechanical_strain_prop_name', requested by 'model_name_face' is not defined on block 0''',
I want to know how to correctly use '''prop_getter_suffix''' parameter.
Lily
